src/app/elevation/qgselevationprofilewidget.cpp

```diff
@@ -890,7 +891,10 @@ void QgsElevationProfileWidget::exportResults( Qgis::ProfileExportType type )
 
   const QList< QgsMapLayer * > layersToGenerate = mCanvas->layers();
   QList< QgsAbstractProfileSource * > sources;
-  sources.reserve( layersToGenerate.size() );
+  QList< QgsAbstractProfileSource * > registrySources = QgsApplication::profileSourceRegistry()->profileSources();
```

Comment:
```suggestion
  const QList< QgsAbstractProfileSource * > registrySources = QgsApplication::profileSourceRegistry()->profileSources();
```

Reply:
Addressed in https://github.com/qgis/QGIS/pull/57311/commits/de9abdfeaa806000a060d22b308048dd245beb75